Client/Frontend/Home/Pocket/PocketDataAdaptor.swift

```diff
@@ -0,0 +1,92 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol PocketDataAdaptor {
+    var onTapAction: ((IndexPath) -> Void)? { get set }
+
+    func getPocketData() -> [PocketStandardCellViewModel]
+}
+
+protocol PocketDelegate: AnyObject {
+    func didLoadNewData()
+}
+
+class PocketDataAdaptorImplementation: PocketDataAdaptor, FeatureFlaggable {
+
+    var notificationCenter: NotificationProtocol
+
+    private let pocketAPI: PocketStoriesProviding
+    private let pocketSponsoredAPI: PocketSponsoredStoriesProviding
+    private let storyProvider: StoryProvider
+    private var pocketStoriesViewModels = [PocketStandardCellViewModel]()
```

Comment:
The adaptor shouldn't know anything about the view models or ideally anything view related. It looks like this model object is mostly constructed from the PocketStory data object so maybe you can store an array of those here and then convert to the view models in the PocketViewModel.